File: DesafioAutomacao/src/test/java/com/qa/blogdoagi/tests/BlogDoAgiTest.java
package com.qa.blogdoagi.tests;

import org.testng.annotations.Test;

import com.qa.blogdoagi.base.BaseTest;
import com.qa.blogdoagi.pages.MainPage;

public class BlogDoAgiTest extends BaseTest {
	
	@Test(priority = 1)
	public void validarPesquisa() {
		MainPage mainPage = new MainPage(page);
		mainPage.clicarBotaoPesquisa();
		mainPage.inserirPalavra();
		mainPage.validarPalavraPesquisada();

	}
	
	@Test(priority = 2)
	public void validarSubMenu() {
		MainPage mainPage = new MainPage(page);
		mainPage.focarMenuEmprestimo();
		mainPage.clicarMenuEmprestimo();
		mainPage.validarMensagemEmprestimo();

	}
	
	@Test(priority = 3)
	public void validarInscricaoEmail() {
		MainPage mainPage = new MainPage(page);
		mainPage.preencherEmail();
		mainPage.clicarBotaoInscrever();
		mainPage.clicarBotaoEntendi();
		mainPage.validarAusenciaPopUpInscricao();
		
	}
	
	@Test(priority = 4)
	public void validarPaginacao() {
		MainPage mainPage = new MainPage(page);
		mainPage.clicarBotaoPaginacao();
		mainPage.validarUrlPaginacao2();
	}
	
	@Test(priority = 5)
	public void validarLinkDownloadAPP() {
		MainPage mainPage = new MainPage(page);
		mainPage.validarBotaoAPPStore();
	}

}

File: Tests/BlogDoAgiTest.cs
using PlaywrightTests.Core;
using PlaywrightTests.Pages;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PlaywrightTests.Tests
{
    public class BlogDoAgiTest : BaseTest
    {

        [Test]
        public async Task ValidarPesquisa()
        {
            MainPage _mainpage = new MainPage(page);
            await _mainpage.ClickBotaoPesquisa();
            await _mainpage.InserirPalavra();
            await _mainpage.ValidarPalavraPesquisada();
        }

        [Test]
        public async Task ValidarSubMenu()
        {
            MainPage _mainpage = new MainPage(page);
            await _mainpage.FocarMenuEmprestimo();
            await _mainpage.ClicarMenuEmprestimo();
            await _mainpage.ValidarMensagemEmprestimo();
        }

        [Test]
        public async Task ValidarInscricaoEmail()
        {
            MainPage _mainpage = new MainPage(page);
            await _mainpage.PreencherEmail();
            await _mainpage.ClicarBotaoInscrever();
            await _mainpage.ClicarBotaoEntendi();
            await _mainpage.ValidarAusenciaPopUpInscricao();
        }

        [Test]
        public async Task ValidarPaginacao()
        {
            MainPage _mainpage = new MainPage(page);
            await _mainpage.ClicarBotaoPaginacao();
            await _mainpage.ValidarURLPaginacao();
        }

        [Test]
        public async Task ValidarLinkDownloadAPP()
        {
            MainPage _mainpage = new MainPage(page);
            await _mainpage.ValidarBotaoAppSotre();
        }

    }
}
